design/deploy.md

```diff
@@ -0,0 +1,98 @@
+# Deploy
+
+A proposal to add support for hosting service deployment to the Polymer CLI.
+
+## Goals
+- Provide new users with a fast way to get a project hosted online.
+- Provide advanced users with an initial bootstrap for more advanced deployments.
+- Demonstrate the [PRPL](https://www.polymer-project.org/2.0/toolbox/server) pattern where HTTP/2 Server Push is available.
+
+## Non-goals
+- Do not try to support every possible deployment environment and configuration, or even a lot of them. Just enough to get users started.
+
+## Design
+
+- Add an `--environment` flag.
+  - Every command knows that it's being run in the context of this environment.
+  - Under the `default` environment, all commands operate as they do today.
+  - Under other environments:
+     - `build` now outputs the boilerplate needed to deploy the project to a given service, in addition to the traditional app build output.
+     - `serve` runs an appropriate HTTP server that emulates the configured hosting environment, instead of polyserve.
+
+- Add an `environments` section to `polymer.json`.
```

Comment:
You use plural here and singular above - which raises the question of whether or not there is only one, or multiple environments. This section seems to describe multiple user-named environments, but I had assumed that we would have environments that come with a name already. Can you clarify a little?